### ISSUE TITLE
Deal with having two copies of u

### ISSUE THREAD

**Issue by lm2612** [ImperialCollegeLondon/AI4PDEs#1]
## Issue
We currently have two tensors: `u` for the main domain and `uu `for the domain plus the halo to include the boundary conditions. Ideally we would like to only have one copy of `u`.

## Possible solutions
* Maybe with `padding='same'` in the `nn.Conv`? 
* and also with `padding_mode='replicate'` ? 
* https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html
